src/main.py

```diff
@@ -252,19 +259,24 @@ def main():
             logger.debug("Done.")
             logger.debug("Received the following data from Elmer/Ice:", data_from_elmer)
 
-            IN["h"] = data_from_elmer["h"]
-            grid["z"] = IN["h"][0].ravel()
-            # TODO add gradient field later
+            # IN['h'] = data_from_elmer['h']
+            # grid['z'] = IN['h'][0].ravel()
```

Comment:
I assume we need to revert this in order to not break existing functionality?

Why did you have to comment it out in the first place? Is this something that is easy to fix or should we add a guard around this part? I.e.

```python
if grid["grid_type"] == GridInputType.ELMERXIOS:
    logger.warning(f"Grid type {grid["grid_type"]} does not support outputting IN["h"] and grid["z"]. Ignoring these fields.")  # todo: find a fix if necessary
else:
    # usual behavior
```
    
```